### ISSUE TITLE
[Firebase_Crashlytics] TimeoutException when launching on Android

### ISSUE THREAD

**Issue by dancamdev** [FirebaseExtended/flutterfire#50]
I've setup crashlytics on both android and ios for my flutter app. 
While it is working flawlessly on iOS, I'm getting a java.util.concurrent.TimeoutException on Android.

This is the stacktrace:

> Launching lib/main.dart on Android SDK built for x86 in debug mode...
> Built build/app/outputs/apk/debug/app-debug.apk.
> E/CrashlyticsCore(27143): Failed to execute task.
> E/CrashlyticsCore(27143): java.util.concurrent.TimeoutException
> E/CrashlyticsCore(27143): 	at java.util.concurrent.FutureTask.get(FutureTask.java:206)
> E/CrashlyticsCore(27143): 	at com.crashlytics.android.core.CrashlyticsBackgroundWorker.submitAndWait(CrashlyticsBackgroundWorker.java:41)
> E/CrashlyticsCore(27143): 	at com.crashlytics.android.core.CrashlyticsController.handleUncaughtException(CrashlyticsController.java:321)
> E/CrashlyticsCore(27143): 	at com.crashlytics.android.core.CrashlyticsController$6.onUncaughtException(CrashlyticsController.java:301)
> E/CrashlyticsCore(27143): 	at com.crashlytics.android.core.CrashlyticsUncaughtExceptionHandler.uncaughtException(CrashlyticsUncaughtExceptionHandler.java:42)
> E/CrashlyticsCore(27143): 	at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:1073)

I've double checked the setup multiple times and it's correct.

**Comment by KholofeloMoyaba** [comment]
I've got the same issue... not sure what to do now. It works well on iOS.

**Comment by martintadashi** [comment]
Same for me, works well on iOS, and throws this exception on Android

**Comment by anztrax** [comment]
ya me too i had the same issue 😞

**Comment by TrustyTechSG** [comment]
Ya, me too, same error

**Comment by AdamSHurwitz** [comment]
This seems similar to the issue shared on StackOverflow, [Crash with ProGuard using Firebase Libraries](https://stackoverflow.com/questions/57010884/crash-with-proguard-using-firebase-libraries). 

This issue includes Crashlytics with the attempted solutions outlined, none of which have resolved the issue.

**Comment by zirplsoftware** [comment]
Same error here

**Comment by AdamSHurwitz** [comment]
I've shared a post to the [Firebase Talk Google group](https://groups.google.com/forum/#!forum/firebase-talk), and submitted a [customer support](https://firebase.google.com/support/troubleshooter/contact) ticket with Firebase. I'd recommend filing a customer support ticket as well to raise attention to this.

I will update this thread once I learn more. 🔎☝🏻

**Comment by Thamizhan** [comment]
I faced the same issue. 

When I integrated Crashlytics in my flutter app it worked fine on both iOS and Android without any issues. But after sometime I upgraded flutter to latest version 1.17.2 and updated some libraries. After that it started giving this error on run console for Android, for iOS its working without any issues. 

I tried to see output from logcat when running on Android, I got two errors one from Facebook library I used https://github.com/darwin-morocho/flutter-facebook-auth, and I got above crashlytics error after that. So I thought the crashlytics error might be just a warning and my app was crashing due to facebook sdk crash. So I disabled that facebook sdk and ran the app in Android without any other changes, It works just fine. no crash or no logs from CrashlyticsCore on the console. 

I'm not sure this is the case for everyone. I just want you to try checking the logs from logcat when your app is crashing and see if that logs contains any other information about real crash on Android. Once you fix that issue your app should work fine I hope.
